src/undo_log/undo_log.c

```diff
@@ -644,13 +635,14 @@ void UndoLog_Free
 (
 	UndoLog log
 ) {
-	// free each undo operation
-	uint count = array_len(log);
-	for (uint i = 0; i < count; i++) {
-		UndoOp *op = log + i;
+	if(log == NULL) return;
+
+	DataBlockIterator *iter = DataBlock_Scan(log);
+	UndoOp *op;
+	while((op = DataBlockIterator_Next(iter, NULL))) {
 		UndoLog_FreeOp(op);
 	}
```

Comment:
I think we can use the `DataBlock` `fpDestructor` to perform the clean up
```c
DataBlock *log = DataBlock_New(32, 32, sizeof(...), UndoLog_FreeOp);
```
In that case this logic can be removed.

Reply:
we cant because I want to be able to free without running the descriptor see the rollback function